airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/source.py

```diff
@@ -104,24 +109,32 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         :param config: A Mapping of the user input configuration as defined in the connector spec.
         """
         args = self.convert_config2stream_args(config)
-        # sorted in alphabet order
-        return [
-            GroupMemberships(**args),
-            Groups(**args),
-            Macros(**args),
-            Organizations(**args),
-            SatisfactionRatings(**args),
-            SlaPolicies(**args),
-            Tags(**args),
-            TicketAudits(**args),
-            TicketComments(**args),
-            TicketFields(**args),
-            TicketForms(**args),
-            TicketMetrics(**args),
-            TicketMetricEvents(**args),
-            Tickets(**args),
-            Users(**args),
-            Brands(**args),
-            CustomRoles(**args),
-            Schedules(**args),
-        ]
+        all_streams_mapping = {
+            # sorted in alphabet order
+            "group_membership": GroupMemberships(**args),
+            "groups": Groups(**args),
+            "macros": Macros(**args),
+            "organizations": Organizations(**args),
+            "satisfaction_ratings": SatisfactionRatings(**args),
+            "sla_policies": SlaPolicies(**args),
+            "tags": Tags(**args),
+            "ticket_audits": TicketAudits(**args),
+            "ticket_comments": TicketComments(**args),
+            "ticket_fields": TicketFields(**args),
+            "ticket_forms": TicketForms(**args),
+            "ticket_metrics": TicketMetrics(**args),
+            "ticket_metric_events": TicketMetricEvents(**args),
+            "tickets": Tickets(**args),
+            "users": Users(**args),
+            "brands": Brands(**args),
+            "custom_roles": CustomRoles(**args),
+            "schedules": Schedules(**args),
+        }
+        # check the users Zendesk Subscription Plan
+        subscription_plan = UserSubscriptionStream(**args).get_subscription_plan()
+        if subscription_plan != FULL_ACCESS_PLAN:
```

Comment:
(pseudocode)
```suggestion
        all_streams = stream_name_map.values()
        if subscription_plan != FULL_ACCESS_PLAN:
            return all_streams - FULL_ACCESS_ONLY_STREAMS
        return all_streams
```

Reply:
We would still need to use the for loop if we have more than 1 element inside of FULL_ACCESS_ONLY_STREAMS.